cloud/amazon/ec2_group_facts.py

```diff
@@ -0,0 +1,186 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: ec2_group_facts
+author: "Stefan Horning (@stefanhorning)"
+version_added: "2.2"
+short_description: Finds exising EC2 security groups and returns their facts
+description:
+    - Finds a list of existing EC2 security groups based on given filters. Returns the facts of each found security group. This module has a dependency on boto3.
+options:
+  name:
+    description:
+      - Name of the security group.
+    required: false
+  group_id:
+    description:
+      - ID of the security group.
+    required: false
+  description:
+    description:
+      - Description of the security group.
+    required: false
+  vpc_id:
+    description:
+      - ID of the VPC the security belongs to.
+    required: false
```

Comment:
The problem is you're making an assumption about what the user wants to filter by and I can guarentee that within a month of the module being published, someone will come along and ask for filtering on something obscure.

That's why i think it's best to build it from the start.

boto3 also does allow you to search separately using sec group names and sec group ids so you could include these two along with the filter parameter.